From 9e0a585e2dd5c835e4917d7a8384a62a591c3811 Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Wed, 2 May 2012 10:40:16 +0530
Subject: [PATCH 0955/1092] ARM: OMAP2+: am335x: add LCD support for EVM-SK

Adding platform resource information for LCD on AM335x EVM-SK.
LCD panel used is Newhaven Display-4.3 of resolution
480 X 272 pixels.

Signed-off-by: Patil, Rachna <rachna@ti.com>
Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 23 +++++++++++++++++++++--
 1 file changed, 21 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 1fb5533..8c7d8eb 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -120,6 +120,12 @@ struct da8xx_lcdc_platform_data TFC_S9700RTWV35TR_01B_pdata = {
 	.type			= "TFC_S9700RTWV35TR_01B",
 };
 
+struct da8xx_lcdc_platform_data  NHD_480272MF_ATXI_pdata = {
+	.manu_name              = "NHD",
+	.controller_data        = &lcd_cfg,
+	.type                   = "NHD-4.3-ATXI#-T-1",
+};
+
 #include "common.h"
 
 #include <linux/lis3lv02d.h>
@@ -909,7 +915,7 @@ out:
 
 static void lcdc_init(int evm_id, int profile)
 {
-
+	struct da8xx_lcdc_platform_data *lcdc_pdata;
 	setup_pin_mux(lcdc_pin_mux);
 
 	if (conf_disp_pll(300000000)) {
@@ -917,9 +923,21 @@ static void lcdc_init(int evm_id, int profile)
 				"register LCDC\n");
 		return;
 	}
+	switch (evm_id) {
+	case GEN_PURP_EVM:
+		lcdc_pdata = &TFC_S9700RTWV35TR_01B_pdata;
+		break;
+	case EVM_SK:
+		lcdc_pdata = &NHD_480272MF_ATXI_pdata;
+		break;
+	default:
+		pr_err("LCDC not supported on this evm (%d)\n",evm_id);
+		return;
+	}
 
-	if (am33xx_register_lcdc(&TFC_S9700RTWV35TR_01B_pdata))
+	if (am33xx_register_lcdc(lcdc_pdata))
 		pr_info("Failed to register LCDC device\n");
+
 	return;
 }
 
@@ -1726,6 +1744,7 @@ static struct evm_dev_cfg evm_sk_dev_cfg[] = {
 	{mmc0_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{rgmii1_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{rgmii2_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
+	{lcdc_init,     DEV_ON_BASEBOARD, PROFILE_ALL},
 	{NULL, 0, 0},
 };
 
-- 
1.7.11.2

